src/natcap/invest/habitat_quality.py

```diff
@@ -651,106 +687,331 @@ def quality_op(degradation, habitat):
-                _RARITY_NODATA)
+    decay_func(max_dist_pixel, kernel_path)
 
-            LOGGER.info('Finished rarity computation on %s land cover.'
-                        % lulc_time)
-    LOGGER.info('Finished habitat_quality biophysical calculations')
 
+def _compare_lucodes_sensitivity(
+        sensitivity_dict, unique_lucode_pickle_lookup):
+    """Compare LULC unique codes to codes listed in dictionary.
 
-def resolve_ambiguous_raster_path(path, raise_error=True):
-    """Determine real path when we don't know true path extension.
+    Parameters:
+        sensitivity_dict (dict): dictionary representing sensitivity for LULC
+            values with a column of 'codes'.
+        unique_lucode_pickle_lookup (list): list of paths to pickle files that
+            represent a python set of unique LULC values for the LULC rasters
+            provided in the model.
+    Returns:
+        None
+    """
+    raster_unique_lucodes = set()
+    for lucode_path in unique_lucode_pickle_lookup:
+        with open(lucode_path, 'rb') as fh:
+            lucode_dict = pickle.load(fh)
+        raster_unique_lucodes.update(lucode_dict['codes'])
+
+    # check if there's any lucode from the LULC rasters missing in the
+    # sensitivity table
+    table_unique_lucodes = set(sensitivity_dict.keys())
+    missing_lucodes = raster_unique_lucodes.difference(table_unique_lucodes)
+    if missing_lucodes:
+        raise ValueError(
+            'The following land cover codes were found in your landcover '
+            'rasters but not in your sensitivity table. Check your '
+            'sensitivity table to see if they are missing: %s. \n\n' %
+            ', '.join([str(x) for x in sorted(missing_lucodes)]))
+
+
+def _collect_unique_lucodes(raster_path_band, pickle_path):
+    """Get unique pixel values from raster and pickle as a Python set.
 
     Parameters:
-        path (string): file path that includes the name of the file but not
-            its extension
+        raster_path_band (tuple): a 2 tuple of the form
+            (filepath to raster, band index).
+        pickle_path (string): a path to output of a pickled Python set.
+    Returns:
+        None
+    """
+    lulc_path = raster_path_band[0]
+    # declare a set to store unique codes from lulc rasters
+    raster_unique_lucodes = set()
 
-        raise_error (boolean): if True then function will raise an
-            ValueError if a valid raster file could not be found.
+    for _, lulc_block in pygeoprocessing.iterblocks(raster_path_band):
+        raster_unique_lucodes.update(numpy.unique(lulc_block))
+
+    # Remove the nodata value from the set of landuser codes.
+    nodata = pygeoprocessing.get_raster_info(lulc_path)['nodata'][0]
+    try:
+        raster_unique_lucodes.remove(nodata)
+    except KeyError:
+        # KeyError when the nodata value was not encountered in the
+        # raster's pixel values.  Same result when nodata value is
+        # None.
+        pass
```

Comment:
you can use `raster_unique_lucodes.discard(nodata)` here so you don't need to wrap up a try/except